src/main/java/org/radarbase/appserver/controller/RadarUserController.java

```diff
@@ -103,6 +99,7 @@ public RadarUserController(
             "The provided token does not have enough privileges.");
       }
     } else {
+      if (forceFcmToken) this.userService.checkFcmTokenExistsAndReplace(userDto);
```

Comment:
The only caveat is that the authorisation code is now not consistent with the operation. One subject who has access to itself only, can now modify the other subjects' token, since permission is only checked on the one subject. But considering firebase token will only be same in case of same device, maybe this is not a concern?

Reply:
Yes that's true and we're also now resetting the token in the active app when the user resets so it will now always be unique. So this should be even rarer in the future.